File: SGT.WebAdmin/ViewsScripts/Enumeradores/EnumDescricaoUsoEmpresaPagamentoEletronico.js
var EnumDescricaoUsoEmpresaPagamentoEletronicoHelper = function () {    
    this.Nenhum = 0;
    this.PagamentoSalario = 1;
    this.PagamentoFerias = 2;
    this.AdiantamentoSalario = 3;
    this.DecimoTerceiroSalario = 4;
    this.AdiantamentoDecimoTerceiroSalario = 5;
    this.BonusMetaVenda = 6;
    this.BonusPerformance = 7;
    this.ComissaoSobreVendas = 8;
    this.ParticipacaoLucrosDaEmpresa = 9;
    this.BonusPorProducao = 10;
    this.DecimoQuartoSalario = 11;
    this.HoraExtra = 12;
    this.GratificacaoPremio = 13;
    this.RescisaoContratual = 14;
    this.AdiantamentoDiarias = 15;
};

EnumDescricaoUsoEmpresaPagamentoEletronicoHelper.prototype = {
    obterOpcoes: function () {
        return [
            { text: "0 - Não informado", value: this.Nenhum },
            { text: "1 - Pagamento de salários", value: this.PagamentoSalario  },
            { text: "2 - Pagamento de férias", value: this.PagamentoFerias },
            { text: "3 - Adiantamento de salário", value: this.AdiantamentoSalario  },
            { text: "4 - 13º Salário", value: this.DecimoTerceiroSalario  },
            { text: "5 - Adiantamento de 13º Salário", value: this.AdiantamentoDecimoTerceiroSalario  },
            { text: "6 - Bônus por metas de vendas", value: this.BonusMetaVenda  },
            { text: "7 - Bônus por performance", value: this.BonusPerformance  },
            { text: "8 - Comissão sobre vendas", value: this.ComissaoSobreVendas  },
            { text: "9 - Participação lucros da empresa", value: this.ParticipacaoLucrosDaEmpresa  },
            { text: "10 - Bônus por produção / serviço", value: this.BonusPorProducao  },
            { text: "11 - 14º Salário", value: this.DecimoQuartoSalario  },
            { text: "12 - Hora extra", value: this.HoraExtra  },
            { text: "13 - Gratificação/ Prêmio", value: this.GratificacaoPremio  },
            { text: "14 - Rescisão contratual", value: this.RescisaoContratual },
            { text: "15 - Adiantamento de diárias", value: this.AdiantamentoDiarias }
        ];
    }
};

var EnumDescricaoUsoEmpresaPagamentoEletronico = Object.freeze(new EnumDescricaoUsoEmpresaPagamentoEletronicoHelper());

File: Dominio/ObjetosDeValor/Embarcador/Enumeradores/EnumDescricaoUsoEmpresaPagamentoEletronico.cs
namespace Dominio.ObjetosDeValor.Embarcador.Enumeradores
{
    public enum DescricaoUsoEmpresaPagamentoEletronico
    {
        Nenhum = 0,
        PagamentoSalario = 1,
        PagamentoFerias = 2,
        AdiantamentoSalario = 3,
        DecimoTerceiroSalario = 4,
        AdiantamentoDecimoTerceiroSalario = 5,
        BonusMetaVenda = 6,
        BonusPerformance = 7,
        ComissaoSobreVendas = 8,
        ParticipacaoLucrosDaEmpresa = 9,
        BonusPorProducao = 10,
        DecimoQuartoSalario = 11,
        HoraExtra = 12,
        GratificacaoPremio = 13,
        RescisaoContratual = 14,
        AdiantamentoDiarias = 15
    }

    public static class DescricaoUsoEmpresaPagamentoEletronicoHelper
    {
        public static string ObterDescricao(this DescricaoUsoEmpresaPagamentoEletronico finalidadePagamento)
        {
            switch (finalidadePagamento)
            {
                case DescricaoUsoEmpresaPagamentoEletronico.Nenhum: return "Nenhum";
                case DescricaoUsoEmpresaPagamentoEletronico.PagamentoSalario: return "PAGAMENTO DE SALARIO";
                case DescricaoUsoEmpresaPagamentoEletronico.PagamentoFerias: return "PAGAMENTO DE FERIAS";
                case DescricaoUsoEmpresaPagamentoEletronico.AdiantamentoSalario: return "ADIANTAMENTO DE SALARIO";
                case DescricaoUsoEmpresaPagamentoEletronico.DecimoTerceiroSalario: return "13o. SALARIO";
                case DescricaoUsoEmpresaPagamentoEletronico.AdiantamentoDecimoTerceiroSalario: return "ADIANTAMENTO 13o. SALARIO";
                case DescricaoUsoEmpresaPagamentoEletronico.BonusMetaVenda: return "BONUS POR METAS DE VENDAS";
                case DescricaoUsoEmpresaPagamentoEletronico.BonusPerformance: return "BONUS POR PERFORMANCE";
                case DescricaoUsoEmpresaPagamentoEletronico.ComissaoSobreVendas: return "COMISSAO SOBRE VENDAS";
                case DescricaoUsoEmpresaPagamentoEletronico.ParticipacaoLucrosDaEmpresa: return "PARTICIPACAO LUCROS DA EMPRESA";
                case DescricaoUsoEmpresaPagamentoEletronico.BonusPorProducao: return "BONUS POR PRODUCAO / SERVICO";
                case DescricaoUsoEmpresaPagamentoEletronico.DecimoQuartoSalario: return "14o. SALARIO";
                case DescricaoUsoEmpresaPagamentoEletronico.HoraExtra: return "HORA EXTRA";
                case DescricaoUsoEmpresaPagamentoEletronico.GratificacaoPremio: return "GRATIFICACAO / PREMIO";
                case DescricaoUsoEmpresaPagamentoEletronico.RescisaoContratual: return "RESCISAO CONTRATUAL";
                case DescricaoUsoEmpresaPagamentoEletronico.AdiantamentoDiarias: return "ADIANTAMENTO DE DIÁRIAS";
                default: return string.Empty;
            }
        }
    }
}
